lib/shared-method.js

```diff
@@ -137,7 +137,9 @@ function SharedMethod(fn, name, sc, options) {
     this.errors = [this.errors];
   }
 
-  this.stringName = (sc ? sc.name : '') + (isStatic ? '.' : '.prototype.') + name;
+  var addPrototypeString = !/^prototype\./.test(name);
+  this.stringName = (sc ? sc.name : '') +
+    (!isStatic && addPrototypeString ? '.prototype.' : '.') + name;
```

Comment:
As far as this patch is concerned, I think we should not add support for `name:'prototype.method'`, as I already mentioned before, so this block should be reverted.

However, in order to prevent other users from running into the situation described in https://github.com/strongloop/strong-remoting/pull/275/files#r51434265, I am proposing to add a check here instead and throw an error when the method name starts with `prototype.`.

```js
if (/^prototype\./.test(name)) {
  var msg = 'Incorrect API usage. Shared methods on prototypes should be ' +
    'created via "new SharedMethod(fn, "name", { isStatic: false })';
  throw new Error(msg);
}